pumpkin/src/world/mod.rs

```diff
@@ -336,9 +336,19 @@ impl World {
         None
     }
 
-    /// Gets a Player by name
-    pub fn get_player_by_name(&self, name: &str) -> Option<Arc<Player>> {
-        // not sure of blocking lock
+    /// Gets a Player by username
+    pub async fn get_player_by_name(&self, name: &str) -> Option<Arc<Player>> {
+        for player in self.current_players.lock().await.values() {
+            if player.gameprofile.name == name {
+                return Some(player.clone());
+            }
+        }
+        None
+    }
+
+    /// Gets a Player by username (Blocking - Legacy use only)
+    pub fn get_player_by_name_blocking(&self, name: &str) -> Option<Arc<Player>> {
+        // TODO: Remove this blocking functions when commands are async.
```

Comment:
Since https://github.com/Snowiiii/Pumpkin/commit/7347bae275e65351ced7f80098e98aab48889aa4 is merged now. You can replace blocking_lock with lock in the other function

Reply:
Sorry, just to double check, did you mean just remove the blocking function altogether?

In the linked commit, it is still using the blocking function [here](https://github.com/Snowiiii/Pumpkin/blob/7347bae275e65351ced7f80098e98aab48889aa4/pumpkin/src/commands/arg_player.rs#L29).

it is also still using the blocking lock in that commit [here](https://github.com/Snowiiii/Pumpkin/blob/7347bae275e65351ced7f80098e98aab48889aa4/pumpkin/src/world/mod.rs#L367).

Please kindly clarify for me :) 

the problematic file is arg_player.rs which is the only reference for the blocking function